packages/editor/src/components/post-actions/actions.js

```diff
@@ -896,47 +884,39 @@ const resetTemplateAction = {
 	hideModalHeader: true,
 	RenderModal: ( { items, closeModal, onActionPerformed } ) => {
 		const [ isBusy, setIsBusy ] = useState( false );
-		const { revertTemplate, removeTemplates } = unlock(
-			useDispatch( editorStore )
-		);
+		const { revertTemplate } = unlock( useDispatch( editorStore ) );
 		const { saveEditedEntityRecord } = useDispatch( coreStore );
 		const { createSuccessNotice, createErrorNotice } =
 			useDispatch( noticesStore );
 		const onConfirm = async () => {
 			try {
-				if ( items[ 0 ].type === TEMPLATE_PART_POST_TYPE ) {
-					await removeTemplates( items );
-				} else {
```

Comment:
@jorgefilipecosta, this special condition was introduced in #61612, but I couldn't find any details on why it was necessary.

Calling `removeTemplates` when resetting a template part from the Site Editor causes a missing entity error.

Reply:
Hi @Mamaduka,

In https://github.com/WordPress/gutenberg/pull/61612 I implemented things this way to replicate the logic that already existed before for template part reset.

Previously when a template part reset happened the template part post of the user was deleted it was always the case so I kept the behavior. 
With this change instead of deleting we keep the user template part and change it to be equal to what the template set. This may cause issues during updates etc.
Previously @youknowriad even said that maybe even for templates during a reset we should just delete the user template too. I'm not sure if going in the direction of keeping the user entity also for template parts is a good idea.

Reply:
Update: In my tests, it seems the template part is still deleted as expected so thins look. I'm checking how it happens given that revertTemplate at least previously did not apply a delete 🤔

Reply:
Checking the code from `wp/6.5` branch. It looks like that the editor sidebar was using the [`revertTemplate`](https://github.com/WordPress/gutenberg/blob/39b57b0c63d91c6dc1b1b9998c441f22b4cdfb61/packages/edit-site/src/components/sidebar-edit-mode/template-panel/template-actions.js#L44) while "DataViews" was using the [`removeTemplates`](https://github.com/WordPress/gutenberg/blob/7ebb55ca27e7bdf6275e73ff75de608158127533/packages/edit-site/src/components/page-patterns/dataviews-pattern-actions.js#L215) action.

The problem with using `removeTemplates` while viewing the same entity in canvas is that it also deletes the entity from the store and requires page refresh to load it.

Here's what happens when I restore old behavior:

https://github.com/WordPress/gutenberg/assets/240569/01db00d1-8f68-48b1-9c91-5f812c5ba5ba

Reply:
On revert templates when we pass source: 'theme', to edit an entity it is in fact an instruction to delete the entity from the database. I will add a comment there to make the code easier to read in the future, I was not aware of that.

But yah we should apply this simplification as we can use the use behavior on both.

Reply:
Thank you, @jorgefilipecosta!